src/Message/Handler/AbstractBusinessCentralHandler.php

```
@@ -0,0 +1,89 @@
+use Sylius\Component\Order\Repository\OrderRepositoryInterface;
+use Symfony\Component\Messenger\Exception\UnrecoverableMessageHandlingException;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Symfony\Component\Workflow\Registry;
+use Symfony\Component\Workflow\WorkflowInterface;
+use Throwable;
+
+abstract class AbstractBusinessCentralHandler implements MessageHandlerInterface
+{
+    protected EntityManagerInterface $orderManager;
+
+    protected OrderRepositoryInterface $orderRepository;
+
+    protected Registry $workflowRegistry;
+
+    public function __construct(
+        EntityManagerInterface $orderManager,
+        OrderRepositoryInterface $orderRepository,
+        Registry $workflowRegistry
+    ) {
+        $this->orderManager = $orderManager;
+        $this->orderRepository = $orderRepository;
+        $this->workflowRegistry = $workflowRegistry;
+    }
+
+    protected function transition(WorkflowInterface $workflow, OrderInterface $order, string $transition): void
+    {
+        if (!$workflow->can($order, $transition)) {
+            throw new UnrecoverableMessageHandlingException(sprintf(
+                'Could not complete ERP state transition "%s". State was: %s', $transition, $order->getErpState()
+            ));
+        }
+
+        $workflow->apply($order, $transition);
+        $this->orderManager->flush();
+    }
+
+    protected function getWorkflow(OrderInterface $order): WorkflowInterface
+    {
+        return $this->workflowRegistry->get($order, OrderErpGraph::NAME);
+    }
+
+    protected static function getErrorFromException(Throwable $e): string
+    {
+        return sprintf(
+            "%s(%s) at %s:%s\n\nTrace:\n%s",
+            get_class($e),
+            $e->getMessage(),
+            $e->getFile(),
+            $e->getLine(),
+            $e->getTraceAsString()
+        );
+    }
+
+    /**
+     * @param mixed|OrderInterface $orderId
+     *
+     * @throws \Safe\Exceptions\StringsException
```

Igor and I have added this to our list of ignored exceptions in phpstorm. You can add the `ExceptionInterface` from Safe :)